### DISCUSSION TITLE
Add files dynamically and upload when done

### DISCUSSION THREAD

**Discussion by AntoninJuquel** [rpldy/react-uploady#346] [Q&A] [answered]
Hello I wanted to create the same behaviour as [WeTransfer](https://wetransfer.com/), which is :
- First you add / remove files you wanna upload
- Then when you are ready you press on the upload button

And in my case it will upload them on my server.

But I didn't get the part where I can manually trigger the upload with a button and upload multiple batch.

Fortunatly I found exactly what I wanted to achieve in the [storybook](https://react-uploady-storybook.netlify.app/?path=/story/ui-upload-preview--with-different-batch-items-method ) but I can't find the exact source code ...

**Comment by yoavniran** [comment, marked as answer]
Hi @AntoninJuquel This has been on my list to demo for a while now, so thanks for the reminder :) 

I created a sandbox showing how to achieve what you described: 
https://codesandbox.io/s/react-uploady-add-and-remove-before-upload-vk1rs5

Its pretty basic and doesnt take into account every edge-case of course but should help get you started. Here's how it looks like: 


https://user-images.githubusercontent.com/1102278/158998108-809cccb3-5b10-41c2-9810-3564d9204cde.mp4


Three's also a guide based on the same example with a bit more explanation here - 
https://react-uploady.netlify.app/docs/guides/AddRemoveThenUpload/
